site/components/src/main/java/uk/nhs/hee/web/beans/PublicationLandingPage.java

```diff
@@ -97,4 +97,19 @@ public List<AssetResource> getLanguageVersionsContent() {
     public HippoBean getFeaturedContentBlock() {
         return getLinkedBean("hee:featuredContentBlock", HippoBean.class);
     }
+
+    @HippoEssentialsGenerated(internalName = "hee:globalTaxonomyHealthcareTopics")
```

Comment:
OOTB, [dynamic bean generation](https://xmdocumentation.bloomreach.com/library/concepts/content-beans/dynamic-content-beans.html) seem to generate getters for taxonomy based fields with `org.onehippo.taxonomy.contentbean.TaxonomyClassification` return type which might be handy for us. For instance, if you remove these taxonomy getters from the bean it will generate one which could readily be used in our freemarker templates:

```
...
<#if document.globalTaxonomyHealthcareTopicsClassification??>
    <#list document.globalTaxonomyHealthcareTopicsClassification.taxonomyValues as category>
        Topic = ${category.key}::${category.label}<br/>
    </#list>
</#if>
...
```

But, we may want to work with these getters in our component classes and want to avoid the devs generating using essentials. And so, I am thinking perhaps we add the following method to `uk.nhs.hee.web.beans.BaseDocument`:
```
public Taxonomy getTaxonomy(final String taxonomyName) {
    final TaxonomyManager taxonomyManager = HstServices.getComponentManager().getComponent(
            TaxonomyManager.class.getSimpleName(), "org.onehippo.taxonomy.contentbean");
    return taxonomyManager.getTaxonomies().getTaxonomy(taxonomyName);
}
```
which could be used by freemarker templates to get all categories of a taxonomy if needed.

And add the following getters to `uk.nhs.hee.web.beans.PublicationLandingPage`:
```
@HippoEssentialsGenerated(internalName = "hee:globalTaxonomyProfessions", allowModifications = false)
public TaxonomyClassification getGlobalTaxonomyProfessions() throws RepositoryException {
    return new TaxonomyClassification(
            this.getNode().getProperty("hee:globalTaxonomyProfessions"),
            getTaxonomy("GlobalProfessions"));
}

@HippoEssentialsGenerated(internalName = "hee:globalTaxonomyHealthcareTopics", allowModifications = false)
public TaxonomyClassification getGlobalTaxonomyHealthcareTopics() throws RepositoryException {
    return new TaxonomyClassification(
            this.getNode().getProperty("hee:globalTaxonomyHealthcareTopics"),
            getTaxonomy("GlobalHealthcareTopics"));
}

@HippoEssentialsGenerated(internalName = "hee:globalTaxonomyPublicationTypes", allowModifications = false)
public TaxonomyClassification getGlobalTaxonomyPublicationTypes() throws RepositoryException {
    return new TaxonomyClassification(
            this.getNode().getProperty("hee:globalTaxonomyPublicationTypes"),
            getTaxonomy("GlobalPublicationTypes"));
}
```
I know we are hardcoding taxonomy names (e.g. `GlobalProfessions`, etc) here, but I guess it is unlikely to change frequently. How does it sound ? thanks

Reply:
This is an excellent observation ! I've added that now and refactored a couple of other classes (Component) to make use of it.